### ISSUE TITLE
decode fail

### ISSUE THREAD

**Issue by zhouchaopku** [ywu40/RaptorCodes_Cplusplus#2]
Hi, ywu40, I have implement these code under vs2013, when I set the parameter to be k = 100, len = 1000 or 800, loss = 5, it will be crash when decode_test() is called. And form the callstack I can see that it crashed at FreeData() Line 47.  The crash only happens for some certain parameters.
By the way, Can you explain how to check the decode is success (I can not find the decode results) when it it decoded successfully.
At last, can you explain how the "Encoded Packet Total Number" is determined. 
Thank you very much.
